ImputationPipeline/Imputation.wdl

```diff
@@ -0,0 +1,475 @@
+    File ref_dict
+    Int disk_size = 400
+  }
+  command <<<
+  zgrep -v "##contig=<ID=" ~{vcf} | bgzip -c > no_contigs.vcf.gz 
+    gatk IndexFeatureFile -F no_contigs.vcf.gz
+    ## update the header of the merged vcf
+    gatk UpdateVCFSequenceDictionary --source-dictionary ~{ref_dict} --output final_call.vcf.gz --replace -V no_contigs.vcf.gz
+  >>>
+  runtime {
+    docker: "us.gcr.io/broad-gatk/gatk:4.1.1.0"
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "8 GiB"
+  }
+  output {
+    File output_vcf = "final_call.vcf.gz"
+    File output_vcf_index = "final_call.vcf.gz.tbi"   
+  }
+}
+
+task RemoveSymbolicAlleles {
+  input {
+    File original_vcf
+    File original_vcf_index 
+    String output_basename = "michigan_allchroms.no_symbolic"
+    Int disk_size = 200
+  }
+  command {
+    gatk SelectVariants -V ~{original_vcf} -xl-select-type SYMBOLIC --select-type-to-exclude MIXED --exclude-non-variants TRUE --remove-unused-alternates TRUE -O ~{output_basename}.vcf.gz
+  }
+  output {
+    File output_vcf = "~{output_basename}.vcf.gz"
+    File output_vcf_index = "~{output_basename}.vcf.gz.tbi"
+  }
+  runtime {
+   # docker: "us.gcr.io/broad-gatk/gatk:latest"
+    docker: "us.gcr.io/broad-gatk/gatk:4.1.1.0"
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "4 GB"
+  }
+}
+
+task SeparateMultiallelics {
+  input {
+    File original_vcf
+    File original_vcf_index 
+    String output_basename = "michigan_allchroms.no_multi_symbolic"
+    Int disk_size = 200
+  }
+  command {
+    bcftools norm -m - ~{original_vcf} | bgzip -c > ~{output_basename}.vcf.gz
+    bcftools index -t ~{output_basename}.vcf.gz
+  }
+  output {
+    File output_vcf = "~{output_basename}.vcf.gz"
+    File output_vcf_index = "~{output_basename}.vcf.gz.tbi"
+  }
+  runtime {
+    docker: "skwalker/imputation:with_python"
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "4 GB"
+  }
+}
+
+task SplitSample {
+  input {
+    File vcf
+    File vcf_index
+    String sample
+    Int disk_size = 400
+  }
+  command {
+    gatk SelectVariants -V ~{vcf} -sn ~{sample} -O ~{sample}.unsorted.vcf
+    gatk --java-options "-Xmx8g" SortVcf -I ~{sample}.unsorted.vcf -O ~{sample}.vcf.gz
+  }
+  runtime {
+    docker: "us.gcr.io/broad-gatk/gatk:4.1.1.0"
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "9 GB"
+  }
+  output {
+    File output_gzipped_vcf = "~{sample}.vcf.gz"
+    File output_gzipped_vcf_index = "~{sample}.vcf.gz.tbi"
+  }
+}
+
+task QConArray {
+  input {
+    File input_vcf
+    File input_vcf_index
+    String output_vcf_basename
+   }
+    Int disk_size = 400
+
+  command <<<
+    # site missing rate < 5% ; hwe p > 1e-6 ; in
+    vcftools --gzvcf ~{input_vcf} --stdout --max-missing 0.05 --hwe 0.000001 --recode -c > tmp.vcf
+    awk 'BEGIN {FS="\t"}; {if ($1 ~ /#/ || ($4=="A" && $5!="T") || ($4=="T" && $5!="A") || ($4=="C" && $5!="G" ) || ($4=="G" || $5!="C")) print $0}' tmp.vcf | bgzip -c > ~{output_vcf_basename}.vcf.gz
```

Comment:
probably want to fix this  too

Reply:
Michigan does this in their QC because they aren't be used in the phasing/imputation algorithms and I wasn't sure if it would break them to give them the data or if they would ignore it properly?

Reply:
Wait I'm sorry.. I misread their documentation, but looking at the code they do allow for ambiguous SNPs

https://github.com/genepi/imputationserver/blob/57445da98cfeff9cb69978b08547d885d1bcccbb/src/main/java/genepi/imputationserver/steps/fastqc/StatisticsTask.java